File: fun_mode.py
"""Fun Mode party pack for The Cube Beta.

Drop this file into the game's add-ons folder and reload add-ons.
"""


def register(api):
    api.about(
        name="Fun Mode: Party Pack",
        version="1.0",
        author="Cube Community",
        description=(
            "Adds colorful party shapes, a floaty Moon Bounce, "
            "and a fast Confetti Cannon event."
        ),
    )

    api.shape(
        name="Disco Ball",
        kind="circle",
        size=32,
        color=(150, 110, 255),
        weight=0.8,
    )
    api.shape(
        name="Confetti Star",
        kind="polygon",
        sides=5,
        size=25,
        color=(255, 75, 170),
        weight=0.65,
    )
    api.shape(
        name="Party Sun",
        kind="polygon",
        sides=8,
        size=38,
        color=(255, 205, 55),
        weight=1.1,
    )

    api.event(
        name="Moon Bounce",
        duration=10,
        wind=180,
        gravity_scale=0.16,
        spawn_count=8,
        banner="Moon Bounce! Everything is light, floaty, and ready to party!",
        color=(125, 155, 255),
    )
    api.event(
        name="Confetti Cannon",
        duration=6,
        wind=1200,
        gravity_scale=0.55,
        spawn_count=12,
        banner="Confetti Cannon! A colorful blast races across the beach!",
        color=(255, 80, 175),
    )

File: addons/fun_mode.rb
require "json"

# Fun Mode's Ruby companion pack. The game reads the JSON printed on the
# final output line.
addon = {
  name: "Fun Mode: Chaos Pack",
  version: "1.1",
  author: "Cube Community",
  description: "Adds silly shapes, playful events, and the shared Fun Mode GUI theme.",
  gui: {
    badge: "FUN MODE  PY + RB",
    panel_title: "FUN METER",
    action: "PARTY EVENT  [F]",
    accent: [255, 70, 220],
    secondary: [65, 235, 205],
    confetti_colors: [
      [255, 70, 220],
      [65, 235, 205],
      [255, 205, 55],
      [105, 90, 245],
      [255, 105, 75]
    ]
  },
  shapes: [
    {
      name: "Jelly Drop",
      kind: "circle",
      size: 39,
      color: [65, 235, 205],
      weight: 0.45
    },
    {
      name: "Party Diamond",
      kind: "polygon",
      sides: 4,
      size: 31,
      color: [255, 70, 220],
      weight: 0.9
    },
    {
      name: "Lucky Hex",
      kind: "polygon",
      sides: 6,
      size: 35,
      color: [85, 230, 105],
      weight: 1.25
    }
  ],
  events: [
    {
      name: "Reverse Rush",
      duration: 7,
      wind: -1150,
      gravity_scale: 0.75,
      spawn_count: 10,
      banner: "Reverse Rush! The party suddenly zooms the other way!",
      color: [255, 105, 75]
    },
    {
      name: "Space Party",
      duration: 11,
      wind: -180,
      gravity_scale: 0.08,
      spawn_count: 7,
      banner: "Space Party! Shapes drift through a nearly weightless world!",
      color: [105, 90, 245]
    }
  ]
}

puts JSON.generate(addon)
